web/src/components/ItemModal.tsx

```diff
@@ -27,15 +37,53 @@ export const ItemModal: React.FC<ItemModalProps> = ({ item, isOpen, onClose }) =
 
   if (!isOpen || (item == null)) return null
 
+  const handleEditClick = (): void => {
+    setIsEditing(true)
+  }
+
+  const handleIncrement = (): void => {
+    setInventory(prev => prev + interval)
+  }
+
+  const handleDecrement = (): void => {
+    setInventory(prev => (prev - interval >= 0 ? prev - interval : 0)) // Ensure inventory doesn't go below 0
+  }
+
+  const handleSave = (): void => {
+    setIsEditing(false)
+    alert(`Save complete. New inventory: ${inventory}`)
+    // You can add your API call here later
+  }
+
   return (
     <div className='modal-overlay'>
       <div className='modal-content'>
         <h2>{item.productName}</h2>
         <img src={item.imageUrl} alt={item.productName} />
         <p>SKU: {item.sku}</p>
-        <p>Total Inventory: {item.totalInventory}</p>
-        <p>Committed: {item.committed}</p>
-        <p>Available: {item.available}</p>
+        <div className='inventory-edit-section'>
+          <label>Total Inventory:</label>
+          {isEditing
+            ? (
+              <div className='edit-inventory-container' style={{ display: 'flex', alignItems: 'center', gap: '10px' }}>
+                <button onClick={handleDecrement}>-{interval}</button>
+                <input
+                  type='number'
+                  value={inventory}
+                  onChange={(e) => setInventory(Number(e.target.value))}
```

Comment:
I'm curious why we're calling `setInventory()` in the onChange here as well as the useEffect. Do we need both?